File: overlay/python/hydra-kernel/src/main/python/hydra/dsl/meta/lib/chars.py
"""Phantom-typed term DSL for the hydra.lib.chars library."""

from hydra.typed import TypedTerm
from hydra.dsl.meta.phantoms import lib_primitive1 as primitive1


def is_alpha_num(c: TypedTerm) -> TypedTerm:
    """Check if a character is alphanumeric."""
    return primitive1(c)


def is_lower(c: TypedTerm) -> TypedTerm:
    """Check if a character is lowercase."""
    return primitive1(c)


def is_space(c: TypedTerm) -> TypedTerm:
    """Check if a character is whitespace."""
    return primitive1(c)


def is_upper(c: TypedTerm) -> TypedTerm:
    """Check if a character is uppercase."""
    return primitive1(c)


def to_lower(c: TypedTerm) -> TypedTerm:
    """Convert a character to lowercase."""
    return primitive1(c)


def to_upper(c: TypedTerm) -> TypedTerm:
    """Convert a character to uppercase."""
    return primitive1(c)

File: overlay/haskell/hydra-kernel/src/main/haskell/Hydra/Dsl/Meta/Lib/Chars.hs
-- | Phantom-typed term DSL for the hydra.lib.chars library

module Hydra.Dsl.Meta.Lib.Chars where

import Hydra.Typed
import Hydra.Dsl.Meta.Phantoms
import qualified Hydra.Dsl.Terms as Terms
import qualified Hydra.Dsl.Prims as Prims
import qualified Hydra.Lib.Chars as DefChars


-- | Check whether a character is alphanumeric.
isAlphaNum :: TypedTerm Int -> TypedTerm Bool
isAlphaNum = primitive1 DefChars.isAlphaNum

-- | Check whether a character is lowercase.
isLower :: TypedTerm Int -> TypedTerm Bool
isLower = primitive1 DefChars.isLower

-- | Check whether a character is a whitespace character.
isSpace :: TypedTerm Int -> TypedTerm Bool
isSpace = primitive1 DefChars.isSpace

-- | Check whether a character is uppercase.
isUpper :: TypedTerm Int -> TypedTerm Bool
isUpper = primitive1 DefChars.isUpper

-- | Convert a character to lowercase.
toLower :: TypedTerm Int -> TypedTerm Int
toLower = primitive1 DefChars.toLower

-- | Convert a character to uppercase.
toUpper :: TypedTerm Int -> TypedTerm Int
toUpper = primitive1 DefChars.toUpper
